### DISCUSSION TITLE
How to use advanced type utilities - Prisma.Payload

### DISCUSSION THREAD

**Discussion by DharanBro** [prisma/prisma#19712] [Q&A] [answered]
I am trying to create a extensions that takes same input as the `model.create`.  After reading [this](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/model#advanced-type-safety-improve-the-type-safety-and-developer-experience-of-your-custom-model-methods) I ended up using 

```
import { Prisma } from '@prisma/client';

type MyCustomInput = Prisma.Payload<ModelType, 'create'>

```

but `MyCustomInput` ends up having the type `unknown`.

Is my understanding wrong? Any help would be appreciated.

**Comment by nurul3101** [comment, marked as answer]
Hello @DharanBro 👋 

> I am trying to create a extensions that takes same input as the model.create

If I understand correctly, can't you just use the `Prisma.ModelCreateInput` type?
For Example, if your model is called User
Can you try using this?
```typescript
import { Prisma } from '@prisma/client';

type MyCustomInput = Prisma.UserCreateInput;
```

**Reply by nurul3101** [reply]
I am checking to see if you had a chance to look at my last response.

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
@DharanBro Do let us know in case you still need any assistance.

**Reply by DharanBro** [reply]
@nurul3101 Thanks for your time. This is what I ended up using it. But still I need to figure out why `Prisma.Payload` was not working.
